OracleTuxedo/core/dockerfiles/buildContainerImage.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+#set -x
+
+
+usage() {
+cat << EOF
+
+Usage: buildContainerImage.sh [-hs] -v version -i installer [-m md5value]
+Builds a Container Image for Oracle Tuxedo.
+  
+Parameters:
+   -h: Help
+   -v: Version to build. Required.
+       Choose one of subdirectories. e.g. 21.1.0.0.0
+   -s: Skips the MD5 check of packages.
+   -i: Installer name. Required.
+   -m: MD5 value expected. Required if -s not specified.
+LICENSE UPL 1.0 
+
+Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+EOF
+exit 0
+}
+
+# Validate packages
+checksumPackages() {
+  echo "Checking if required packages are present and valid..."
+  if [ -z "$MD5VALUE" ]; then   # -m option is a MUST
+    echo "You must specify $INSTALLER MD5 value with -m option"
+    exit
+  fi
+  MD5="${MD5VALUE}  ${VERSION}/${INSTALLER}"
+  echo "MD5 =  ${MD5}"
+  MD5_CHECK="$(md5sum "${VERSION}"/"${INSTALLER}")"
+  echo "MD5_CHECK =  ${MD5_CHECK}"
+
+  if [ "$MD5" != "$MD5_CHECK" ]
+  then
+    echo "MD5 does not match! Download again!"
+    exit
+  fi
+}
+
+if [ "$#" -eq 0 ]; then usage; fi
+echo "====================="
+#
+
+VERSION=
+INSTALLER=
+MD5VALUE=
+SKIPMD5=0
+while getopts "hsi:m:v:" optname; do
+  case "$optname" in
+    "h")
+      usage
+      ;;
+    "s")
+      SKIPMD5=1
+      ;;
+    "i")
+     INSTALLER="$OPTARG"
+      ;;
+    "m")
+     MD5VALUE="$OPTARG"
+      ;;
+    "v")
+      VERSION="$OPTARG"
+      ;;
+    *)
+    # Should not occur
+      echo "Unknown error while processing options inside buildContainerImage.sh"
+      ;;
+  esac
+done
+
+if [ -z "$VERSION" ]; then   # -v option is a MUST
+    echo "You must specify Tuxedo version with -v option"
+    exit
+fi
+
+if [ -z "$INSTALLER" ]; then   # -i option is a MUST
+    echo "You must specify installer name with -i option"
+    exit
+fi
+
+
+if [ ! "$SKIPMD5" -eq 1 ]; then
+  checksumPackages
+else
+  echo "Skipped MD5 checksum."
+fi
+
+echo "====================="
+
+if [ ! -e "${VERSION}"/"${INSTALLER}" ]
+then
+  echo "Download the Tuxedo ZIP Distribution and"
+  echo "drop the file ${INSTALLER} in ${VERSION} folder before"
+  echo "building this Tuxedo container image!"
+  exit 
+fi
+
+# Proxy settings
+PROXY_SETTINGS=""
+# shellcheck disable=SC2154
+if [ "${http_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg http_proxy=${http_proxy}"
+fi
+
+# shellcheck disable=SC2154
+if [ "${https_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg https_proxy=${https_proxy}"
+fi
+
+# shellcheck disable=SC2154
+if [ "${ftp_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg ftp_proxy=${ftp_proxy}"
+fi
```

Comment:
"docker build" uses the ftp_proxy in case the client sets it. Perhaps the client has the images on a ftp server.

Reply:
Build time environment variables are not persisted into the image, so even if they did have this setting, it wouldn't be available to a container built from this image.

Reply:
Maybe I am missing something here. Let's say we have the below command in one of our Dockerfiles:

`FROM oracle/serverjre:8`

Then when we run "docker build" it will try to pull this image. But what if this image is hosted on a FTP server and a FTP proxy is needed to access this FTP server ?

Reply:
This variable is not used by the container engine to pull the base image, it's used by the commands running inside the build environment. Thus, even if you set an FTP_PROXY to pull an image, you don't need to make that available to the build environment. So this is not required.

Reply:
Okay, will remove these env vars and push the changes soon.